Update Rust dependencies

cbgbt opened the pull request:
**Description of changes:**

```
tools: update tough to 0.12
tools: cargo update

sources: update nix to 0.23
sources: update tough to 0.12
sources: update to actix-web-4.0.0-beta.10
sources: cargo update
```
Not updated: [gptman](https://github.com/bottlerocket-os/bottlerocket/issues/1620), [hyper](https://github.com/bottlerocket-os/bottlerocket/issues/1471) (in `sources/`), [x509-parser](https://github.com/bottlerocket-os/bottlerocket/issues/1743)
webpki (must match the version used by reqwest)

**Testing done:**

- [x] Unit tests pass.
- [x] Built AWS k8s 1.21 for aarch64 and ECS AMIs for x86_64. Confirmed `systemctl status` `running`, dmesg and journal OK. Ran a pod/task OK. Made API changes OK.
- [x] Ensured that handlebars templating changes don't cause semantic changes to rendered templates (thanks @tjkirch)
- [x] Confirmed that `cargo make repo` and `cargo make ssm` still work OK.
- [x] Test `apiclient exec` with the new `nix` change

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

cbgbt committed:
tools: update tough to 0.12

cbgbt committed:
tools: cargo update

cbgbt committed:
sources: update nix to 0.23

cbgbt committed:
sources: update tough to 0.12

cbgbt committed:
sources: update actix-web to 4.0.0-beta.10

cbgbt committed:
sources: cargo update

webern commented on sources/api/migration/migrator/src/main.rs:
```diff
@@ -479,10 +490,10 @@ where
 
 fn load_manifest(repository: &tough::Repository) -> Result<Manifest> {
-    let target = "manifest.json";
+    let target = "manifest.json".try_into().unwrap();
```
Did you forget to use an error variant here?

cbgbt replied:
Because `"manifest.json"` is a constant, we know it can't possibly return an error here, so we just went with `unwrap()`. Should apply to any other addition of unwrap in this PR.

cbgbt replied:
I ended up adding `context()` calls to these constant unwraps, which should make clippy lints more meaningful, which I think is worth it.

webern commented on sources/deny.toml:
```diff
@@ -52,8 +52,11 @@ skip = [
 
-    # older version used by tough 0.11.2
-    { name = "serde_plain", version = "0.3.0" },
+    # older version used by webpki-roots 0.21.1
```
I looked at the lock file and I don't see a change to the webpki-roots, which is a good thing because of https://github.com/bottlerocket-os/bottlerocket/blob/5a083d917e0c1101bec5a8668f5a5a8048239454/sources/webpki-roots-shim/Cargo.toml#L3..L11

cbgbt replied:
Yeah. In this case, one of our other deps (I believe `actix-web`) moved forward to use a newer version of `pem`. Our cargo-deny config flags differences in dependency versions, but we can't change webpki-roots due to what you've linked.

webern commented on sources/updater/updog/src/main.rs:
```diff
@@ -583,10 +590,10 @@ fn main_inner() -> Result<()> {
 
 fn load_manifest(repository: &tough::Repository) -> Result<Manifest> {
-    let target = "manifest.json";
+    let target = "manifest.json".try_into().unwrap();
```
Another error variant is needed here.

Also I might suggest using the type name so a reader can see what type is involved.

```rust
let target = TargetName::new("manifest.json").context(error::InvalidTargetName{ blah blah }?;
```

webern commented on tools/pubsys/src/repo.rs:
```diff
@@ -352,10 +352,11 @@ where
     match repo_load_result {
         // If we load it successfully, build an editor and manifest from it.
         Ok(repo) => {
+            let target = "manifest.json".try_into().unwrap();
```
Same as other locations.

BTW, if we get clippy going in Bottlerocket CI https://github.com/bottlerocket-os/bottlerocket/issues/1776 then we can add this to make sure we don't panic anywhere:

```rust
#![deny(
    clippy::expect_used,
    clippy::get_unwrap,
    clippy::panic,
    clippy::panic_in_result_fn,
    clippy::panicking_unwrap,
    clippy::unwrap_in_result,
    clippy::unwrap_used
)]
```

tjkirch replied:
I wonder if tough could have const checks for static strings, so we could construct things like this without runtime error checking...

tjkirch reviewed (commented):
LGTM if the remaining testing passes.

cbgbt commented:
The previous two force pushes:
* Added the `context()` calls where previously `unwrap()` was being used
* Rebased atop the latest code in the `develop` branch.

Based on test failures, looks like I need to modify the `nix` change to also update for `apiclient exec`, which was added in the rebase. I'll get a fix ready.

cbgbt commented:
I've pushed just a change to move `apiclient` to `nix 0.23`. I tested `apiclient exec` with the changes.

tjkirch reviewed (approved):
LGTM assuming checks pass!
